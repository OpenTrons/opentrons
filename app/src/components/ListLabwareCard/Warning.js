// @flow
import * as React from 'react'

import { Icon } from '@opentrons/components'
import styles from './styles.css'

export type WarningProps = {|
  children: React.Node,
|}

export function Warning(props: WarningProps): React.Node {
  const { children } = props

  return (
    <div className={styles.warning}>
      <Icon className={styles.warning_icon} name="alert-circle" />
      <div>{children}</div>
    </div>
  )
}
