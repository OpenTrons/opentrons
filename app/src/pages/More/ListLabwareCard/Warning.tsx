import * as React from 'react'

import { Icon } from '@opentrons/components'
import styles from './styles.css'

export interface WarningProps {
  children: React.ReactNode
}

export function Warning(props: WarningProps): JSX.Element {
  const { children } = props

  return (
    <div className={styles.warning}>
      <Icon className={styles.warning_icon} name="alert-circle" />
      <div>{children}</div>
    </div>
  )
}
