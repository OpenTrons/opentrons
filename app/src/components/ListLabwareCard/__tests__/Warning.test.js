// @flow
import * as React from 'react'
import { shallow } from 'enzyme'

import { Warning } from '../Warning'

describe('Warning', () => {
  test('component renders', () => {
    const wrapper = shallow(<Warning>AH!!!</Warning>)

    expect(wrapper).toMatchSnapshot()
  })
})
