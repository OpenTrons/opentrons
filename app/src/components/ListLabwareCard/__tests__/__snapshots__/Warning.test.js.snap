// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning component renders 1`] = `
<div
  className="warning"
>
  <Icon
    className="warning_icon"
    name="alert-circle"
  />
  <div>
    AH!!!
  </div>
</div>
`;
